src/lib.rs

```diff
@@ -190,21 +192,15 @@ impl DocData {
             // example:: -> example
             name: root_def.qualname[..(name_len - 2)].to_string(),
             docs: root_def.docs.clone(),
-            modules: Vec::new(),
+            metadata: Vec::new(),
         };
 
-        let data = DocData::build_data(config, root_id, &mut krate)?;
+        DocData::build_data(config, root_id, &mut krate)?;
 
-        Ok(DocData { krate, data })
+        Ok(DocData { krate })
     }
 
-    fn build_data(
-        config: &Config,
-        root_id: analysis::Id,
-        krate: &mut Crate,
-    ) -> Result<HashMap<String, Item>> {
-        let mut data = HashMap::new();
-
+    fn build_data(config: &Config, root_id: analysis::Id, krate: &mut Crate) -> Result<()> {
```

Comment:
@steveklabnik I go back and forth on that. As I said before, the main reason passing the crate in felt weird to me was that that seemed to signal that this function's main focus should be on the crate. Now that it is all about the crate, I don't feel this is too weird, though the name should definitely be changed to reflect this new role too.